types/serverless/plugins/aws/provider/awsProvider.d.ts

```diff
@@ -368,7 +368,7 @@ declare namespace Aws {
     interface Sns {
         topicName: string;
         displayName?: string;
-        filterPolicy?: string[] | { [key: string]: string };
+        filterPolicy?: string[] | { [key: string]: string[] };
```

Comment:
filterPolicy is an object of attributes to be filtered against string, numeric or boolean value. It can also contain exists, anything-but... filters. (cf https://docs.aws.amazon.com/sns/latest/dg/sns-subscription-filter-policies.html)
If you agree, I'd rather go for the `filterPolicy?: Record<string, unknown>` type